src/Storages/MergeTree/MergeTreePartition.cpp

```diff
@@ -467,6 +467,42 @@ void MergeTreePartition::create(const StorageMetadataPtr & metadata_snapshot, Bl
     }
 }
 
+void MergeTreePartition::createAndValidateMinMaxPartitionIds(const StorageMetadataPtr & metadata_snapshot, Block block_with_min_max_partition_ids, ContextPtr context)
+{
+    if (!metadata_snapshot->hasPartitionKey())
+        return;
+
+    auto partition_key_names_and_types = executePartitionByExpression(metadata_snapshot, block_with_min_max_partition_ids, context);
+    value.resize(partition_key_names_and_types.size());
+
+    /// Executing partition_by expression adds new columns to passed block according to partition functions.
+    /// The block is passed by reference and is used afterwards. `moduloLegacy` needs to be substituted back
+    /// with just `modulo`, because it was a temporary substitution.
+    static constexpr auto modulo_legacy_function_name = "moduloLegacy";
+
+    size_t i = 0;
+    for (const auto & element : partition_key_names_and_types)
+    {
+        auto & partition_column = block_with_min_max_partition_ids.getByName(element.name);
+
+        if (element.name.starts_with(modulo_legacy_function_name))
+            partition_column.name = "modulo" + partition_column.name.substr(std::strlen(modulo_legacy_function_name));
```

Comment:
If I am not mistaking, calling `sizeof` as is would return the variable type size. In order to do this, I believe we need to do something like `static constexpr const char modulo_legacy_function_name[] = "moduloLegacy";`. That's ok for me, will do it.

Btw, `strlen` is able to compute this at compile time with optimizations on. See https://godbolt.org/z/jqq4cP9xv.

Reply:
Actually, can't we just use a `std::string_view` and do `modulo_legacy_function_name.size()`?

Reply:
`std::string::traits_type::length` can also be used in compile time (it's constexpr since C++17):

https://en.cppreference.com/w/cpp/string/char_traits/length

Reply:
Sure, but string_view is ok, right?